### ISSUE TITLE
Panic on admittedly silly axiom

### ISSUE THREAD

**Issue by philzook58** [mwillsey/egg-smol#72]
https://www.mwillsey.com/egg-smol/?program=XQAAgACeAAAAAAAAAAAUGQgnfMUD9dO10BC2jbJotgLClFTWhuxd3X1pNTX5W8B1pnvqFbP-Z9e9Nklrmkcj2koM9emIop4mcnfahUjjHhdqJLXm1KNuDIX1T2pDuh93M7OwPZsIlGSZ0J0jVlzUUL5wnBvhQ8MXkradYNkf_9rAEAA%253D

```
(datatype KAT
    (A)
    (par KAT KAT)
)

(rewrite (par p p) p)
(rule ((= r (par q r)) (= q (par q r))) ((union r q)))

; tests
(define q (par A A))
(run 10)
```

```
[INFO ] database size: 10
[INFO ] Made 1 updates
thread 'main' panicked at 'assertion failed: i < self.len()', src/unionfind.rs:65:9
stack backtrace:
```

**Comment by oflatt** [comment]
Fixed by #98

**Comment by yihozhang** [comment]
We can close this one @mwillsey
